sync-core/src/main/java/com/cloudant/sync/query/Index.java

```diff
@@ -0,0 +1,165 @@
+//  except in compliance with the License. You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package com.cloudant.sync.query;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class provides functionality to manage an index
+ */
+public class Index {
+
+    private static final Logger logger = Logger.getLogger(Index.class.getName());
+
+    public static final String JSON_TYPE = "json";
+
+    public static final String TEXT_TYPE = "text";
+
+    private static final List<String> validTypes = Arrays.asList(JSON_TYPE, TEXT_TYPE);
+
+    private static final String TOKENIZE = "tokenize";
+
+    private static final String DEFAULT_TOKENIZER = "simple";
+
+    private static final List<String> validSettings = Arrays.asList(TOKENIZE);
+
+    protected final List<Object> fieldNames;
+
+    protected final String indexName;
+
+    protected final String indexType;
+
+    protected final Map<String, String> indexSettings;
+
+    private Index(List<Object> fieldNames,
+                  String indexName,
+                  String indexType,
+                  Map<String, String> indexSettings) {
+        this.fieldNames = fieldNames;
+        this.indexName = indexName;
+        this.indexType = indexType;
+        this.indexSettings = indexSettings;
+    }
+
+    public static Index setUp(List<Object> fieldNames, String indexName) {
+        // JSON index type is the default
+        return setUp(fieldNames, indexName, JSON_TYPE);
+    }
+
+    public static Index setUp(List<Object> fieldNames, String indexName, String indexType) {
+        return setUp(fieldNames, indexName, indexType, null);
+    }
+
+    /**
+     * The setUp method handles index specific validation and ensures that the constructed
+     * Index object is valid.
+     *
+     * @param fieldNames the field names in the index
+     * @param indexName the index name
+     * @param indexType the index type (json or text)
+     * @param indexSettings the optional settings used to configure the index.
+     *                      Only supported parameter is 'tokenize' for text indexes only.
+     * @return the Index object or null if arguments passed in were invalid.
+     */
+    public static Index setUp(List<Object> fieldNames,
+                              String indexName,
+                              String indexType,
```

Comment:
I agree with your point on enumeration however, this `Index` class is meant to, in part, be a way to declutter the `IndexManager`, `IndexCreator`, and `IndexUpdater` code.  At present the IndexManager public methods are our interface to the end user.  The IndexManager `ensureIndexed` methods allow for the end user to provide an indexType of String.  In the future it might make sense to lock this down to an enum.  For now I'd like to leave it as it is and revisit this later.